File: bun-hono/src/middleware/auth.middleware.ts
import { Context, Next } from 'hono'
import JwtUtils from '../utils/jwt.utils'
import { appLog } from '../config/logger.pino'

export default class AuthMiddleware {
  static async authenticate(c: Context, next: Next) {
    const authHeader = c.req.header('Authorization')
    if (!authHeader?.startsWith('Bearer ')) {
      return c.json({ message: 'Unauthorized' }, 401)
    }
    const token = authHeader.split(' ')[1]
    try {
      const payload = await JwtUtils.verifyToken(token)
      if (!payload) {
        return c.json({ message: 'Unauthorized' }, 401)
      }
      appLog.debug("Auth Called")
      appLog.debug({payload})
      c.set('userId', payload.id)
      c.set('email', payload.email)
      await next()
    } catch (err) {
      console.error('JWT verification error:', err)
      return c.json({ message: 'Unauthorized' }, 401)
    }
  }
}

File: go-fiber/pkg/middleware/auth_middleware.go
package middleware

import (
	"github.com/dist-r/rcontacts-rest/go-fiber/pkg/utils"
	"github.com/gofiber/fiber/v2"
)

func AuthMiddleware(c *fiber.Ctx) error {
	token := c.Get("Authorization")
	if token == "" {
		return fiber.ErrUnauthorized
	}
	//  Remove bearer
	token = token[7:]

	claims, err := utils.VerifyToken(token)
	if err != nil {
		return fiber.ErrUnauthorized
	}
	c.Locals("userID", claims.UserID)
	c.Locals("userEmail", claims.Email)
	return c.Next()
}
